frontend/src/pages/projects/screens/detail/data-connections/ManageDataConnectionModal.tsx

```diff
@@ -76,52 +76,66 @@ const ManageDataConnectionModal: React.FC<ManageDataConnectionModalProps> = ({
+      const promiseActions: Promise<K8sResourceCommon>[] = [];
+
+      if (existingData) {
+        if (AWSDataChanged) {
+          promiseActions.push(replaceSecret(assembledSecret, { dryRun: dryRun }));
+        }
+
+        const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
+          removedConnections.includes(notebook.metadata.name),
+        );
+
+        promiseActions.push(
+          ...notebooksToDisconnect.map((notebook) =>
+            replaceNotebookSecret(
+              notebook.metadata.name,
+              projectName,
+              getSecretsFromList(notebook).filter(({ secretRef: { name } }) => name !== secretName),
+              { dryRun: dryRun },
+            ),
+          ),
+        );
+      } else {
+        promiseActions.push(createSecret(assembledSecret, { dryRun: dryRun }));
       }
 
-      const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
-        removedConnections.includes(notebook.metadata.name),
+      const notebooksToConnect = allAvailableNotebooks.filter((notebook) =>
+        addedConnections.includes(notebook.metadata.name),
       );
+
       promiseActions.push(
-        ...notebooksToDisconnect.map((notebook) =>
-          replaceNotebookSecret(
+        ...notebooksToConnect.map((notebook) =>
+          attachNotebookSecret(
             notebook.metadata.name,
             projectName,
-            getSecretsFromList(notebook).filter(({ secretRef: { name } }) => name !== secretName),
+            secretName,
+            hasEnvFrom(notebook),
+            { dryRun: dryRun },
           ),
         ),
       );
-    } else {
-      promiseActions.push(createSecret(assembledSecret));
-    }
-
-    const notebooksToConnect = allAvailableNotebooks.filter((notebook) =>
-      addedConnections.includes(notebook.metadata.name),
-    );
 
-    promiseActions.push(
-      ...notebooksToConnect.map((notebook) =>
-        attachNotebookSecret(notebook.metadata.name, projectName, secretName, hasEnvFrom(notebook)),
-      ),
-    );
+      if (promiseActions.length > 0) {
+        setErrorMessage('');
+        setIsProgress(true);
+        return await Promise.all(promiseActions)
+          .then(() => Promise.resolve())
+          .catch((e) => {
+            setErrorMessage(e.message || 'An unknown error occurred');
+            setIsProgress(false);
+            return Promise.reject();
+          });
+      } else {
+        return Promise.reject();
+      }
+    };
 
-    if (promiseActions.length > 0) {
-      setErrorMessage('');
-      setIsProgress(true);
-      Promise.all(promiseActions)
-        .then(() => onBeforeClose(true))
-        .catch((e) => {
-          setErrorMessage(e.message || 'An unknown error occurred');
-          setIsProgress(false);
-        });
-    } else {
-      onBeforeClose(false);
-    }
+    runPromiseActions(true).then(() => runPromiseActions(false).then(() => onBeforeClose(true)));
```

Comment:
There is no catch for when you reject -- I haven't tested this yet, but wouldn't that just swallow the error?

Reply:
It would which is intended. As a catch is essentially means "dont run it again". And the handling for the errors are done in the catch statement of the Promise.all

Reply:
Hmm, I guess that's fair -- that was the logic before you got here. Okay 👍